internal/controller/prefixclaim_controller.go

```diff
@@ -75,7 +79,101 @@ func (r *PrefixClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	/* 1. check if the matching Prefix object exists */
+	/* 1. compute and assign the parent prefix if required */
+	// The current design will use prefixClaim.Status.ParentPrefix for storing the selected parent prefix,
+	// and as the source of truth for future parent prefix references
+	if prefixClaim.Status.SelectedParendPrefix == "" /* parent prefix not yet selected/assigned */ {
+		if prefixClaim.Spec.ParentPrefix != "" {
+			prefixClaim.Status.SelectedParendPrefix = prefixClaim.Spec.ParentPrefix
+
+			// set status, and condition field
+			msg := fmt.Sprintf("parentPrefix is provided in CR: %v", prefixClaim.Status.SelectedParendPrefix)
+			if err := r.SetConditionAndCreateEvent(ctx, prefixClaim, netboxv1.ConditionParentPrefixSelectedTrue, corev1.EventTypeNormal, msg); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else if len(prefixClaim.Spec.ParentPrefixSelector) > 0 {
+			// we first check if a prefix can be restored from the netbox
+
+			// since the parent prefix is not part of the restoration hash computation
+			// we can quickly check to see if the prefix with the restoration hash is matched in NetBox
+			h := generatePrefixRestorationHash(prefixClaim)
+			canBeRestored, err := r.NetboxClient.RestoreExistingPrefixByHash(h)
+			if err != nil {
+				return ctrl.Result{Requeue: true}, nil
```

Comment:
Here the log message/condition update is missing in case of an error.

Reply:
Added as requested! Can you see if this is what you are thinking about :)
Thanks